cluster_tools/tests/test_all.py

```diff
@@ -350,37 +362,35 @@ class DummyEnum(Enum):
     PEAR = 2
 
 
-def enum_consumer(value: DummyEnum) -> None:
+def enum_consumer(value: DummyEnum) -> DummyEnum:
     assert value == DummyEnum.BANANA
+    return value
 
 
-def test_cloudpickle_serialization() -> None:
-    enum_consumer_inner = enum_consumer
-
-    for fn in [enum_consumer, enum_consumer_inner]:
-        try:
-            with cluster_tools.get_executor("test_pickling") as executor:
-                _fut = executor.submit(fn, DummyEnum.BANANA)
-            assert fn == enum_consumer
-        except Exception:  # noqa: PERF203 `try`-`except` within a loop incurs performance overhead
-            assert fn != enum_consumer
-
-    assert True
-
-
-def test_map_to_futures_with_debug_sequential() -> None:
-    with cluster_tools.get_executor("debug_sequential") as exc:
-        durations = [4, 1]
+@pytest.mark.parametrize(
+    "executor_key", ["multiprocessing_with_pickling", "sequential_with_pickling"]
+)
+def test_pickling(
+    executor_key: Union[
+        Literal["multiprocessing_with_pickling"], Literal["sequential_with_pickling"]
+    ],
+) -> None:
+    with cluster_tools.get_executor(executor_key) as executor:
+        future = executor.submit(enum_consumer, DummyEnum.BANANA)
+        assert future.result() == DummyEnum.BANANA
```

Comment:
the old test was introduced together with cloudpickle support (see [here](https://github.com/scalableminds/cluster_tools/commit/a5715e9a1a28720d7b029e042ac110f9fbc1d7f5)). unfortunately, it was written weirdly and without comments. since the cloudpickle support was already removed, I'd be fine with removing the entire test (there were some oddities about enum serialization with cloudpickle which this test was supposed to test somehow).
the new test reads much nicer and probably doesn't hurt to keep, though. however, I'd rename it to `test_enum_pickling` :)

Reply:
I'll rename and leave it in as a regression test